llvm/lib/Target/RISCV/MCTargetDesc/RISCVMCTargetDesc.cpp

```diff
@@ -195,7 +195,7 @@ class RISCVMCInstrAnalysis : public MCInstrAnalysis {
     }
     case RISCV::AUIPC:
       setGPRState(Inst.getOperand(0).getReg(),
-                  Addr + (Inst.getOperand(1).getImm() << 12));
+                  Addr + ((int32_t)Inst.getOperand(1).getImm() << 12));
```

Comment:
Use `Addr + SignExtend64<32>(Inst.getOperand(1).getImm() << 12))`. It's more readable.